packages/apollo-cli/src/commands/codegen/generate.ts

```diff
@@ -0,0 +1,209 @@
+          return;
+      }
+    }
+
+    const apiKey = process.env.ENGINE_API_KEY || flags.key;
+    const pullFromEngine = !!apiKey;
+
+    const tasks = new Listr([
+      {
+        title: "Scanning for GraphQL queries",
+        task: async (ctx, task) => {
+          const paths = await globby(flags.queries ? [ flags.queries ] : []);
+          task.title = `Scanning for GraphQL queries (${paths.length} found)`;
+          ctx.queryPaths = paths;
+        }
+      },
+      {
+        title: pullFromEngine ? "Loading schema from Apollo Engine" : "Loading GraphQL schema",
+        task: async ctx => {
+          if (pullFromEngine) {
+            const variables = {
+              id: getIdFromKey(apiKey as string),
+              tag: flags.tag || "current",
+            }
+
+            const engineSchema = await toPromise(
+              execute(engineLink, {
+                query: SCHEMA_QUERY,
+                variables,
+                context: {
+                  headers: { ["x-api-key"]: apiKey },
+                  ...(flags.engine && { uri: flags.engine }),
+                },
+              })
+            );
+
+            if (engineSchema.data && engineSchema.data.service.schema) {
+              ctx.schema = buildClientSchema(
+                engineSchema.data.service.schema
+              );
+            } else {
+              this.error("Failed to get schema from Apollo Engine");
+            }
+          } else {
+            const schemaFileContent = await promisify(fs.readFile)(path.resolve(flags.schema as string));
+            const schemaData = JSON.parse(schemaFileContent as any as string);
+            ctx.schema = buildClientSchema(
+              (schemaData.data) ? schemaData.data : (
+                schemaData.__schema ? schemaData : { __schema: schemaData }
+              )
+            )
+          }
+        }
+      },
+      {
+        title: "Generating query files",
+        task: async (ctx, task) => {
+          task.title = `Generating query files with '${inferredTarget}' target`;
+          generate(
+            ctx.queryPaths,
+            ctx.schema,
+            args.output as string,
+            flags.only ? path.resolve(flags.only) : "",
+            inferredTarget,
+            flags.tagName as string,
+            {
+              passthroughCustomScalars: flags.passthroughCustomScalars || flags.customScalarsPrefix,
+              customScalarsPrefix: flags.customScalarsPrefix || "",
+              addTypename: flags.addTypename,
+              namespace: flags.namespace,
+              operationIdsPath: flags.operationIdsPath,
+              generateOperationIds: !!flags.operationIdsPath,
+              mergeInFieldsFromFragmentSpreads: flags.mergeInFieldsFromFragmentSpreads,
+              useFlowExactObjects: flags.useFlowExactObjects,
+              useFlowReadOnlyTypes: flags.useFlowReadOnlyTypes
+            }
+          )
+        },
+      },
+    ]);
+
+    return tasks.run();
```

Comment:
It would be great for this output to give information about what was generated. Something like `x types, x operations generated`. I'm not sure what the best result would be (does `generate` return anything). In practice its always a win if the CLI shows what it just did